Лаб. 4 6412 Шестериков Борис

BarMan555 opened the pull request:
**ФИО:** Шестериков Борис Владимирович
**Номер группы:** 6412  
**Номер лабораторной:** 4  
**Номер варианта:** 54 
**Краткое описание предметной области:** Поликлиника
**Краткое описание добавленных фич:** В ходе лабораторной работы спроектирован и внедрён сервис автоматической генерации пациентов, обменивающийся данными с основным сервером посредством потокового gRPC-канала. Вся система, включая генератор данных оркестрируется через .NET Aspire

alxmcs commented on DispatcherService.Domain/Repositories/DriverRepository.cs:
```diff
@@ -0,0 +1,69 @@
+
+namespace DispatcherService.Domain.Repositories;
+
+public class DriverRepository : IRepository<Driver>
```
Здесь и далее по коду:
Что-то я не вижу, чтобы имплементации репозиториев потребляли контекст базы данных

alxmcs commented on DispatcherService.Domain/Repositories/DriverRepository.cs:
```diff
@@ -0,0 +1,69 @@
+
+public class DriverRepository : IRepository<Driver>
+{
+    private readonly List<Driver> _drivers = new();
```
У тебя есть список сущностей

alxmcs commented on DispatcherService.Domain/Repositories/DriverRepository.cs:
```diff
@@ -0,0 +1,69 @@
+    /// <param name="id">Идентификатор водителя.</param>
+    /// <returns>Возвращает true, если водитель удалён успешно, иначе false.</returns>
+    public bool Delete(int id)
+    {
+        var driver = GetById(id);
+
+        if (driver == null)
+            return false;
+
+        _drivers.Remove(driver);
+        return true;
```
И ты им по всякому манипулируешь

alxmcs commented on DispatcherService.Domain/Repositories/DriverRepository.cs:
```diff
@@ -0,0 +1,69 @@
+    /// </summary>
+    /// <param name="entity">Объект Driver для добавления.</param>
+    /// <returns>Добавленный объект Driver.</returns>
+    public Driver? Post(Driver entity)
+    {
+        _drivers.Add(entity);
+        return entity;
+    }
```
Только я не вижу ,чтобы данные персистились в базу

alxmcs commented on DispatcherService.Domain/DispatcherServiceContext.cs:
Контекст у тебя используется только в двух местах - миграциях и инжекте в di-контейнер

А имплементация доменных служб у тебя как была во 2 лабе на инмемори коллекциях, так и осталась. В базу ничего не сохраняется

alxmcs commented on DispatcherService.API/Program.cs:
```diff
@@ -0,0 +1,54 @@
+
+var builder = WebApplication.CreateBuilder(args);
+
+// ���������� ������������
```
Тут опять, кстати, приуныла кодировка

BarMan555 committed:
Make sln and 2 prj(classes and tests)

Make solution and 2 project. First for classes of models, second for tests

BarMan555 committed:
Make full structur for PolyclinicSolution

Make full structur for PolyclinicSolution

BarMan555 committed:
Rename projects and folders + add required for properties

Polyclinic -> Hospital

BarMan555 committed:
Make xUnit tests and make fixtures data

BarMan555 committed:
Add summary

BarMan555 committed:
Add extra summary

BarMan555 committed:
Rename Folders

BarMan555 committed:
Cleap up code, remake specialization object, rename fun in tests

BarMan555 committed:
Add DateTime.Now in test

BarMan555 committed:
Add patient interface repository

Create fold Repositories for interfaces of repositories. Add patient interface repository

BarMan555 committed:
Add infrastructure project with InMemoryPatientRepository.cs repository

Add infrastructure project with InMemoryPatientRepository.cs repository

BarMan555 committed:
Add seeder for Patient Repository

Add seeder for Patient Repository

BarMan555 committed:
Add application project.

Add application project: Patient DTO and Patient Service class

BarMan555 committed:
Add interface for doctor and appointnent repositories

Add interface for doctor and appointnent repositories

BarMan555 committed:
Add realization for doctor and appointment repository.

Add realization for doctor and appointment repository.

BarMan555 committed:
Make seeders for doctor and appointment repositories.

Make seeders for doctor and appointment repositories.

BarMan555 committed:
Add DTO class for doctor and appointment classes.

Add DTO class for doctor and appointment classes.

BarMan555 committed:
Add service class for doctor and appointment repositories.

Add service class for doctor and appointment repositories.

BarMan555 committed:
Make repositories fixture for testing sevices.

Make repositories fixture for testing sevices.

BarMan555 committed:
Relocataion fixture folder to test project.

Relocataion fixture folder to test project.

BarMan555 committed:
Add function for working with data and for testing.

Add function for working with data and for testing.

BarMan555 committed:
Add file with repositories and services tests.

Add file with repositories and services tests.

BarMan555 committed:
Add interfaces of seeder and repository. Add abstract class with realization of repository interface.

Make ISeeder.cs, which include basic entity for all other seeders. Make IRepository.cs, which include basic entity for all others repositories. Make InMemoryRepository.cs abstract class which implements basich method from IRepository.cs. Make classes InMemoryPatientRepository.cs, InMemoryDoctorRepository.cs and InMemoryAppointmentRepository.cs inherteds to abstract class InMemoryRepository.cs. Delete IPatientRepository.cs, IDoctorRepository.cs and IAppointmentRepository.cs becaouse they don't need. Classes PatientSeeder.cs, DoctorSeeder.cs and AppointmentSeeder.cs implement ISeeder.cs

BarMan555 committed:
Add project for services contracts

BarMan555 committed:
Make project with API Web Core interface.

Make project with API Web Core interface.

BarMan555 committed:
Add to api project.

Add to api project.

BarMan555 committed:
Make interfacies for sevice classes.

Make interfacies for sevice classes. IApplicationService.cs fro CRUD functions. ILibraryAnalyticsService.cs for analytics functions.

BarMan555 committed:
Add auto mapping in projects.

Add auto mapping in projects.

BarMan555 committed:
Add small fixes.

Add small fixes.

BarMan555 committed:
Add service which implements analytics interface.

Add service which implements analytics interface.

BarMan555 committed:
Add summary.

Add summary.

BarMan555 committed:
Add summary.

Add summary.

BarMan555 committed:
Add summary.

Add summary.

BarMan555 committed:
Why not?

BarMan555 committed:
Make README.md for my project.

Make README.md for my project.

BarMan555 committed:
Fix README.md

BarMan555 committed:
Fix README.md

BarMan555 committed:
Fix README.md

BarMan555 committed:
Update .gitignore

BarMan555 committed:
Add own interface for each service. Add new api methods.

BarMan555 committed:
Fix Dtos

BarMan555 committed:
Fix Seeders

BarMan555 committed:
Enable documentation

BarMan555 committed:
Fix controller and add mapper in Program.cs

BarMan555 committed:
Fix Api controllers.

BarMan555 committed:
Update .gitignore

BarMan555 committed:
Add GetDtos and CreateUpdateDtos. Update controllers and services

BarMan555 committed:
Fix dependecies

BarMan555 committed:
Fix dtos

BarMan555 committed:
Refix services

BarMan555 committed:
Fix analytics methods

BarMan555 committed:
Add async to IRepository.cs

BarMan555 committed:
Make MongoClient.cs and classes for database

BarMan555 committed:
Rename

BarMan555 committed:
Add async (Task<TEntity>)

BarMan555 committed:
Add new projects(Aspire, EF MongoDB)

BarMan555 committed:
Add EFCore project

BarMan555 committed:
Add async

BarMan555 committed:
Add xml

BarMan555 committed:
Add Aspire

BarMan555 committed:
Add Async

BarMan555 committed:
Add aspire project. Add DbContext. Make async repository and not async repository

BarMan555 committed:
Update controller. Delete Logging function

BarMan555 committed:
Small fixing

BarMan555 committed:
Delete useless

BarMan555 committed:
Clear api

BarMan555 committed:
Update Dtos

BarMan555 committed:
Add ObjectID

BarMan555 committed:
Fixing for ObjectId

BarMan555 committed:
Add responce 204

BarMan555 committed:
Add xml

BarMan555 committed:
Add xml and fix appointments rep

BarMan555 committed:
Add, Delete projects

BarMan555 committed:
Delete Logging function

BarMan555 committed:
Modeled database in MongoDB

BarMan555 committed:
Make DataInitializer.cs for database

BarMan555 committed:
Fix seeder

BarMan555 committed:
Delete useless file

BarMan555 committed:
Make MongoDb tests

BarMan555 committed:
Fixing for ObjectId

BarMan555 committed:
Add github action

BarMan555 committed:
Update README.md

BarMan555 committed:
Small fix

BarMan555 committed:
Update .gitignore

BarMan555 committed:
Update .gitignore

BarMan555 committed:
Update .gitignore =\

BarMan555 committed:
Clean up useless usings

BarMan555 committed:
Sort usings

BarMan555 committed:
Relocate manual mapping into MappingProfile.cs

BarMan555 committed:
Rename IRepositoryAsync.cs to IRepository.cs

BarMan555 committed:
Add xml

BarMan555 committed:
Add waiting

BarMan555 committed:
Small fix for better perfomence

BarMan555 committed:
Fix mistake

BarMan555 committed:
Small fix

BarMan555 committed:
Update Specialization.cs and DbContext

BarMan555 committed:
Update id generation

BarMan555 committed:
Fix

BarMan555 committed:
Add Include and Reference to EfCore Repository

BarMan555 committed:
Last fix in 3 lab

BarMan555 committed:
Add grpc contracts and server

BarMan555 committed:
Add grpc to AppHost and Api

BarMan555 committed:
Fix small bugs

alxmcs commented on Hospital/.idea/.idea.Hospital/.idea/.gitignore:
Эти файлы в репозитории явно лишние

alxmcs commented on Hospital/Hospital.Api/Grpc/PatientConsumerService.cs:
```diff
@@ -0,0 +1,97 @@
+    IConfiguration config) : BackgroundService
+{
+    private readonly string _generatorUrl = config["services:generator:grpc:0"] 
+                                            ?? throw new InvalidOperationException("URL генератора не найден в конфигурации");
```
Я бы назвал это скорее `KeyNotFoundException`

alxmcs commented on Hospital/Hospital.Api/Grpc/PatientConsumerService.cs:
```diff
@@ -0,0 +1,97 @@
+                        BloodType = response.BloodType,
+                        RhFactor = response.RhFactor,
+                        PhoneNumber = response.PhoneNumber
+                    };
```
Кстати, раз у тебя уже притащен автомаппер в проект, ты мог бы сгенерировать маппинг его средствами для этой ситуации

alxmcs commented on Hospital/Hospital.Grpc.Server/Services/HospitalGeneratorService.cs:
```diff
@@ -0,0 +1,89 @@
+        }
+    }
+
+    /// <summary>
+    /// Метод генерации пациентов.
+    /// </summary>
```
Вообще, по заданию предполагалась генерация приемов, а не пациентов, ну да ладно, можно оставить и так

alxmcs reviewed (changes requested):
Пара мелочей

BarMan555 committed:
Add AutoMapper in PatientConsumerService.cs

BarMan555 committed:
Update cache .gitignore
